web/tests/stream_data.test.cjs

```diff
@@ -2303,58 +2413,79 @@ run_test("can_toggle_subscription", ({override}) => {
     assert.equal(stream_data.can_toggle_subscription(social), true);
 });
 
-run_test("can_archive_stream", ({override}) => {
+run_test("can_access_stream_email", ({override}) => {
     const social = {
-        subscribed: false,
-        color: "red",
         name: "social",
         stream_id: 2,
-        is_muted: false,
-        invite_only: false,
-        history_public_to_subscribers: false,
-        can_add_subscribers_group: me_group.id,
-        can_administer_channel_group: nobody_group.id,
-        can_subscribe_group: me_group.id,
+        is_archived: false,
+        can_send_message_group: {
+            direct_subgroups: [],
+            direct_members: [me.user_id],
+        },
+    };
+    const bot = {
+        is_active: true,
+        user_id: 999,
+    };
+    const bot_user = {
+        email: "bot@zulip.com",
+        full_name: "Bot User",
+        user_id: 999,
+        is_bot: true,
     };
+    people.init();
+    people.add_active_user(me);
+    people.add_active_user(bot_user);
+    override(bot_data, "get_all_bots_for_current_user", () => [bot]);
     override(current_user, "user_id", me.user_id);
 
-    override(current_user, "is_admin", true);
     social.is_archived = true;
-    assert.equal(stream_data.can_archive_stream(social), false);
-
+    assert.equal(stream_data.can_access_stream_email(social), false);
     social.is_archived = false;
-    assert.equal(stream_data.can_archive_stream(social), true);
 
-    override(current_user, "is_admin", false);
-    assert.equal(stream_data.can_archive_stream(social), false);
+    page_params.is_spectator = true;
+    assert.equal(stream_data.can_access_stream_email(social), false);
+    page_params.is_spectator = false;
 
-    social.can_administer_channel_group = me_group.id;
-    assert.equal(stream_data.can_archive_stream(social), true);
+    social.can_send_message_group.direct_members = [me.user_id];
+    assert.equal(stream_data.can_access_stream_email(social), true);
+
+    social.can_send_message_group.direct_members = [bot.user_id];
+    assert.equal(stream_data.can_access_stream_email(social), true);
+
+    social.can_send_message_group.direct_members = [123];
+    assert.equal(stream_data.can_access_stream_email(social), false);
+
+    override(current_user, "is_guest", true);
+    assert.equal(stream_data.can_access_stream_email(social), false);
```

Comment:
How are you deciding which tests are part of the accest_test_cases  and which are not